### PR TITLE
Added test and comments to my feature

### PR DESCRIPTION
**Pull request by NicolasDuciaume**
This Branch was simply to add my Test and comments. Although they should have been updated within the main branch the tests where still sufficient for what was found yet these are updated comments and I added further tests to allow for an in depth look at the results of specific scenarios that might arise.

### CODE REVIEW

File: src/test/java/SYSC6/Project/UserControllerTest.java [hunk 1]
```diff
@@ -57,59 +57,36 @@ void getUserByUsernameNotExists() throws Exception{
...
-        this.mockMvc.perform(get("/rest/api/user/get/Test/Check")).andExpect(status().isOk()).andExpect(content().json("{\"id\":3,\"username\":\"Test\",\"password\":\"Check\",\"role\":\"FREE_USER\"}"));
-    }
-
-
-    @Test
-    @Order(8)
     void createUser() throws Exception {
         JSONObject obj = new JSONObject();
-        obj.put("username","Nick");
+        obj.put("username","Moe");
```
Comment (Calacorm) [inline src/test/java/SYSC6/Project/UserControllerTest.java:62]: Lol

**Review by Calacorm — APPROVED** [review]
I love everything that is happening here

File: src/test/java/SYSC6/Project/UserControllerTest.java (hunk 1 above)
Comment (NicolasDuciaume) [reply src/test/java/SYSC6/Project/UserControllerTest.java:62]: had to change it to something, users are supposed to have different usernames so the get by username command was causing issues

File: src/main/java/SYSC6/Project/user/UserController.java [hunk 2]
```diff
@@ -84,6 +87,12 @@
         return new ResponseEntity<>(user,HttpStatus.OK);
     }
 
+    /**
+     * Method is used to take in user details and find if that user is an admin of basic user then creates
+     * a new account within the database accordingly
+     * @param userRequest The User details
+     * @return a new user account
+     */
```
Comment (nazifatanzim) [inline src/main/java/SYSC6/Project/user/UserController.java:95]: appreciate the javadoc

**Review by nazifatanzim — APPROVED** [review]

**Pull request merged**